chainguard-images: Add TLS networking requirements (#2442)
chainguard-dev/edu: content/chainguard/chainguard-images/network-requirements.md

Many of the hosts listed on this page use multiple DNS A records or CNAME aliases. Additionally, many A records have a short time to live of 60 seconds, and the majority are less than an hour (3600s).

If your network filters traffic based on IP addresses, ensure that any firewalls update their rules at an appropriate interval to match the TTL for each DNS record.

## Minimum TLS requirements

For guaranteed connectivity, the following TLS requirements must be at

> minimum supported by clients/servers communicating with Chainguard

Reviewer: ```suggestion
minimum supported by clients and servers communicating with Chainguard
```

Reviewer: I know servers are basically clients in this context so you could also leave "servers"  out

Author: We have 3 connectivity cases

* customer client connecting to -> [chainguard container as a server]
* chainguard container client connecting to -> [customer hosted server]
* customer clients connecting -> [chainguard host endpoints cgr.dev etc]

In above pairs, both sides are all unique, and these days all languages actually have different ciphersuites they offer and support as client and as server.

For example go supports TLSv1.0 & TLSv1.1 as a client; but not as a server.
Similarly OpenSSL too has configs to expose different ciphers as a client and as a server.

So it is a mess.

Reviewer: Yeah .. I am just aware of it a bit more in the Java side.. but in the wording here we can therefore stick to "clients and servers"

Containers and endpoints:

> - TLSv1.3 with TLS_AES_256_GCM_SHA384 cipher suite

Reviewer: ```suggestion
- TLSv1.3 with the TLS_AES_256_GCM_SHA384 cipher suite
```

- TLSv1.2 with
  - ECDHE-ECDSA-AES256-GCM-SHA384 cipher string
  - [RFC 7627](https://datatracker.ietf.org/doc/html/rfc7627) Extended Master Secret Extenstion support
- Signatures using P-256 with SHA-256
- Signatures using RSA-PSS with 2048 bits and SHA-256

> The above configuration can be approximately tested with OpenSSL client like so:

Reviewer: ```suggestion
The requirements can be approximately tested with the following OpenSSL client command:
```

```
openssl s_client -cipher @SECLEVEL=2:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384 -ciphersuites TLS_AES_256_GCM_SHA384 -groups P-256 -connect HOST:PORT

> ```

Reviewer: empty line

> > Note, in case of TLSv1.2 connectivity, one has to check the output for `Extended master secret: yes`

Reviewer: ```suggestion
> Note that in the case of TLSv1.2 connectivity you must check the output for `Extended master secret: yes`.
```
